src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs

```
@@ -0,0 +1,273 @@
+                ProcessBlueprintsFromAllManagementGroups();
+            }
+            else
+            {
+                if (Name == null)
+                {
+                    ProcessBlueprintsFromManagementGroup(ManagementGroupName[0]);
+                }
+                else
+                {
+                    ProcessNamedBlueprints(ManagementGroupName[0]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handle the "ByVersion" parameter set.
+        /// </summary>
+        /// <remarks>
+        /// The "ByVersion" parameter set filters published blueprints by version name.
+        /// The "Name" parameter is required for this parameter set.
+        /// </remarks>
+        private void HandleByVersionParameterSet()
+        {
+            foreach (var name in Name)
+            {
+                try
+                {
+                    var bp = BlueprintClient.GetPublishedBlueprint(ManagementGroupName[0], name, VersionName);
+                    WriteObject(bp);
+                }
+                catch (Exception ex)
+                {
+                    if (Name.Length == 1)
+                        WriteExceptionError(ex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handle the "LatestPublished" parameter set.
+        /// </summary>
+        /// <remarks>
+        /// The "LatestPublished" parameter set filters published blueprints by latest published date.
+        /// </remarks>
+        private void HandleLatestPublishedParameterSet()
+        {
+            foreach (var name in Name)
+            {
+                try
+                {
+                    var blueprint = BlueprintClient.GetLatestPublishedBlueprint(ManagementGroupName[0], name);
+
+                    if (blueprint != null)
+                        WriteObject(blueprint);
+                }
+                catch (Exception ex)
+                {
+                    if (Name.Length == 1)
```

See comment above about silent failures. And braces.

---

Fixed.